packages/pds/src/migrate-script/index.ts

```diff
@@ -0,0 +1,417 @@
+import axios from 'axios'
+import * as ui8 from 'uint8arrays'
+import AtpAgent from '@atproto/api'
+import * as plcLib from '@did-plc/lib'
+import SqlRepoStorage from '../sql-repo-storage'
+import { createDeferrable } from '@atproto/common'
+import { envToCfg, envToSecrets, readEnv } from '../config'
+import AppContext from '../context'
+import { FailedTakedown, MigrateDb, Status, TransferPhase, getDb } from './db'
+
+type PdsInfo = {
+  id: number
+  did: string
+  url: string
+  agent: AtpAgent
+}
+
+export const runScript = async () => {
+  const db = await getDb('/data/migrate.db')
+  const env = readEnv()
+  const cfg = envToCfg(env)
+  const secrets = envToSecrets(env)
+  const ctx = await AppContext.fromConfig(cfg, secrets)
+  const adminToken = ui8.toString(
+    ui8.fromString(`admin:${secrets.adminPassword}`, 'utf8'),
+    'base64pad',
+  )
+  const pdsRes = await ctx.db.db.selectFrom('pds').selectAll().execute()
+  const pdsInfos = pdsRes.map((row) => ({
+    id: row.id,
+    did: row.did,
+    url: `https://${row.host}`,
+    agent: new AtpAgent({ service: `https://${row.host}` }),
+  }))
+  const todo = await db
+    .selectFrom('status')
+    .where('status.phase', '<', 7)
+    .where('failed', '!=', 1)
+    .selectAll()
+    .execute()
+  let pdsCounter = 0
+  for (const status of todo) {
+    let pdsId = status.pdsId
+    if (!pdsId) {
+      pdsId = pdsCounter % pdsInfos.length
+      pdsCounter++
+    }
+    const pdsInfo = pdsInfos[pdsId]
```

Comment:
This currently uses the pds id and the index of the pds in the `pdsInfos` array interchangeably, thought they could differ from each other.

Reply:
oh lol could call